src/state/workload.rs

```diff
@@ -621,6 +646,10 @@ impl WorkloadStore {
                 .or_default()
                 .insert(w.uid.clone());
         }
+
+        // We have stored a newly inserted workload, notify watchers
+        // (if any) to wake.
+        self.insert_notifier.send_replace(());
```

Comment:
On a large XDS update, this is going to trigger N times which will then wake up all watchers and have them do a lookup (lookup is cheap, but locks so its not that cheap). Essentially we have an `O(UpdateSize*Waiters)` operation which could be not great.

I don't think this is a big deal in practice, though. But it would be better to operate on the xds message granularity vs per-resource

Reply:
Yeah I gestured at sending the Workload thru the channel as a future optimization, to avoid a hard lookup after insert, can do that if it creates problems - but yeah this is uncommon to begin with.